hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -270,7 +273,9 @@ public void startScanningForJobs() {
                 // If there is no master context and job result at the same time, it means this is the first submission
                 jobSubmitted.inc();
                 jobRepository.putNewJobRecord(jobRecord);
-
+                if (isMaster() && jobConfig.getTimeoutMillis() > 0) {
```

Comment:
The default value is `-1` but we check for `> 0`. Should not it be `> -1`?

Reply:
So this is a bit of a twofold thing, first of all we don't want to accept `0` as a timeout value because we might as well not start the job at all and `-1` is of course the default `no-timeout` value. Additionally we don't want to accept negative values below `-1` either since that doesn't make any sense.

This check would be more expressive as `jobConfig.getTimeoutMillis() != NO_TIMEOUT && jobConfig.getTimeoutMillis() > 0`, but for the sake of brevity I went with plain `> 0`. 

Maybe it would be better to have a `hasTimeout()` method call here instead.

Reply:
Right, but if someone specifies `timeout = 0` we don't schedule any task which results in same behaviour as `-1`. Maybe `0` should be the default? Then there is no confusion around the _0_ gap - I saw `0` being treated quite often as infinite timeout.